src/System.Text.Formatting/System/Text/Formatting/Formatters/SequenceFormatter.cs

```diff
@@ -108,5 +109,7 @@ void IBufferWriter.Advance(int bytes)
 
             _totalWritten += bytes;
         }
+
+        public int MaxBufferSize { get; } = Int32.MaxValue;
```

Comment:
What's the contract of MaxBufferSize? i.e. I don't think consumers of this implementation can call GetSpan with int.MaxValue and expect it to work. In general, are we happy with the MaxBufferSize property being added to IBufferWriter? I have second thoughts about it.

@ahsonkhan, @GrabYourPitchforks, @joshfree

Reply:
MaxBufferSize is to be used only in `Math.Min(required, MaxBufferSize)` expression.

Reply:
On the other hand, it's nice to have sensible default sometimes.

Reply:
if required > MaxBufferSize, the required buffer will not be created -> bug
If required < MaxBufferSize, you don't need MaxBufferSize.

Reply:
What's the semantics of MemoryPool.MaxBufferSize? This is pretty much the same.

Reply:
I don't understand this comment. 
What I am arguing is that I think the property is useless: there are lots of examples where we return int.MaxValue, yet the caller won't be able to get int.MaxValue-sized buffers. And so, the property just creates an illusion of robust code, while in reality there are failures lurking all over the place.

Reply:
I'm saying that MemoryPool<byte> abstraction has exactly same property with exactly same semantics and if we are fixing things both should be fixed.

Reply:
I am not sure the same applies to pools. Pools do know what's the max size buffer they can provide. The problem with IBufferWriter is that not all implementations use pooled memory, and some implementations dont know their max size.